crates/vim/src/motion.rs

```diff
@@ -1274,7 +1274,16 @@ fn wrapping_right(map: &DisplaySnapshot, mut point: DisplayPoint, times: usize)
 }
 
 fn wrapping_right_single(map: &DisplaySnapshot, mut point: DisplayPoint) -> DisplayPoint {
-    let max_column = map.line_len(point.row()).saturating_sub(1);
+    let eol_point = map.clip_point(
+        DisplayPoint::new(point.row(), map.line_len(point.row())),
+        Bias::Left,
+    );
+    let mut eol_chars = map.buffer_chars_at(eol_point.to_offset(map, Bias::Left));
+    let eol_char_len = match eol_chars.next() {
+        Some((ch, _)) => ch.len_utf8() as u32,
+        None => 1,
+    };
+    let max_column = map.line_len(point.row()).saturating_sub(eol_char_len);
```

Comment:
@mastion Thanks! I think this could probably be done without reading the buffer text

Does it work to move the point forward and clip it unconditionally. We know that in the case that +1 clip::right returns ourselves, we're at the end of the line: 
```
let next_point = point;
  *next_point.column_mut() += 1;
    next_point = map.clip_point(next_point, Bias::Right);
if next_point == point {
  if next_point.row == map.max_point() {
    return next_point
  } else {
     return Point::new(next_point.row + 1, 0)
  }     
}
```

If not, we can use your technique; but I think `max_column` is just `eol_point.column` (or maybe `movement::left(eol_point).column`?)